src/probnum/filtsmooth/timeseriesposterior.py

```diff
@@ -36,9 +36,30 @@ class TimeSeriesPosterior(abc.ABC):
         Posterior random variables.
     """
 
-    def __init__(self, locations: np.ndarray, states: np.ndarray) -> None:
-        self.locations = np.asarray(locations)
-        self.states = _randomvariablelist._RandomVariableList(states)
+    def __init__(
+        self,
+        locations: Optional[Iterable[FloatArgType]] = None,
+        states: Optional[Iterable[randvars.RandomVariable]] = None,
+    ) -> None:
+        self._locations = list(locations) if locations is not None else []
+        self._states = list(states) if states is not None else []
+
+    def append(
+        self,
+        location: FloatArgType,
+        state: randvars.RandomVariable,
+    ) -> None:
+        # TODO: Type-checks here?
```

Comment:
Is this TODO still open?

Reply:
Yep.
Technically, we should check the types here, because it wouldn't throw an error if somebody appends all kidns of stuff to those lists.
But this method is probably invoked very often during filtering. So, I wanted to raise the question whether or not we want type checks here?
Safety vs. speed, yet again.

Reply:
What do you think, @mmahsereci , @pnkraemer ?

Reply:
I think its fine like this. An extra type check seems a bit unnecessary to me...

Reply:
Ok thanks to both of you :)